src/platform/user/authentication/utilities.js

```diff
@@ -75,12 +68,25 @@ function redirectWithGAClientId(redirectUrl) {
   }
 }
 
+export function standaloneRedirect() {
+  const searchParams = new URLSearchParams(window.location.search);
+  const application = searchParams.get('application');
+  const to = searchParams.get('to');
+  let url = externalRedirects[application] || null;
+
+  if (url && to) {
+    url = url.replace(new RegExp('/$'), '');
+    url += `/${to.replace(new RegExp('^/'), '').replace('\r\n', '')}`;
+  }
+  return url;
+}
+
 function redirect(redirectUrl, clickedEvent) {
   // Keep track of the URL to return to after auth operation.
   // If the user is coming via the standalone sign-in, redirect to the home page.
   const returnUrl =
     window.location.pathname === '/sign-in/'
-      ? window.location.origin
+      ? standaloneRedirect()
```

Comment:
If `standaloneRedirect` returns null (e.g. if no application param is passed, or an un-configured application param), should this fall back to `window.location.origin` (i.e. back to homepage)?

Reply:
We could be explicit about it here, but I believe that's what AuthApp does if the value is undefined/null/empty.  https://github.com/department-of-veterans-affairs/vets-website/blob/5b01d8ee1b1c70a9cf39f8fef8019d4fb6508930/src/applications/auth/containers/AuthApp.jsx#L131-L139

Reply:
Confirmed in localhost that passing a nonsense application name like "foobar" caused redirect to `http://localhost:3001/auth/login/callback/null`. I'm going to work on a fix